src/ucs/sys/string.h

```diff
@@ -262,19 +262,34 @@ size_t ucs_string_count_char(const char *str, char c);
 size_t ucs_string_common_prefix_len(const char *str1, const char *str2);
 
 
+/*
+ * Return the common parent directory, without the trailing slash.
+ * For example, for /sys/ab12 and /sys/ab23 return /sys
+ *
+ * @param  path1        String pointing to first path.
+ * @param  path2        String pointing to second path.
+ * @param  common_path  Buffer to hold the common parent path.
+ * @param  max_length   Size of the common parent path buffer.
```

Comment:
wht is the use case for trimming the common path buffer? Maybe better fail instead? Or print some warning/diag at least